File: C++/Examples/LED/LED.cpp
#include <Navio2/Led_Navio2.h>
#include <Navio+/Led_Navio.h>
#include <Common/Util.h>
#include <unistd.h>
#include <memory>


std::unique_ptr <Led> get_led()
{
    if (get_navio_version() == NAVIO2)
    {
        auto ptr = std::unique_ptr <Led>{ new Led_Navio2() };
        return ptr;
    } else
    {
        auto ptr = std::unique_ptr <Led>{ new Led_Navio() };
        return ptr;
    }
}

int main(int argc, char *argv[])
{

    if (check_apm()) {
        return 1;
    }

    if (getuid()) {
    
        fprintf(stderr, "Not root. Please launch like this: sudo %s\n", argv[0]);
    }


    auto led = get_led();
    if (!led->initialize())
        return EXIT_FAILURE;

    while (true) {
        led->setColor(Colors::Green);
        printf("LED is green\n");
        sleep(1);

        led->setColor(Colors::Cyan);
        printf("LED is cyan\n");
        sleep(1);

        led->setColor(Colors::Blue);
        printf("LED is blue\n");
        sleep(1);

        led->setColor(Colors::Magenta);
        printf("LED is magenta\n");
        sleep(1);

        led->setColor(Colors::Red);
        printf("LED is red\n");
        sleep(1);

        led->setColor(Colors::Yellow);
        printf("LED is yellow\n");
        sleep(1);
        }

    return 0;
}

File: Python/LED.py
import time
import os

import navio.Common.util
if navio.Common.util.get_navio_version() == "NAVIO2":
    import navio.Navio2.Led as Led
else:
    import navio.Navio.Led as Led


if (os.getuid() != 0):
    print "Not root. Please, launch like this: sudo python LED.py"
    exit(-1)

navio.Common.util.check_apm()

led = Led()

led.setColor('Yellow')
print "LED is yellow"
time.sleep(1)

while (True):

    led.setColor('Green')
    print "LED is green"
    time.sleep(1)

    led.setColor('Cyan')
    print "LED is cyan"
    time.sleep(1)

    led.setColor('Blue')
    print "LED is blue"
    time.sleep(1)

    led.setColor('Magenta')
    print "LED is magenta"
    time.sleep(1)

    led.setColor('Red')
    print "LED is red"
    time.sleep(1)

    led.setColor('Yellow')
    print "LED is yellow"
    time.sleep(1)

